source/integrations/whatsapp.rst

```diff
@@ -0,0 +1,138 @@
+
+
+1. Register a sandbox
+^^^^^^^^^^^^^^^^^^^^^
+
+Login or create a `new Twilio account <https://www.twilio.com`_.
+
+
+Next, make sure you `register the sandbox and confirm it on WhatsApp<https://www.twilio.com/console/sms/whatsapp/learn>`_.
+
+
+2. Building the bot
+^^^^^^^^^^^^^^^^^^^
+
+You can import `WhatsAppBotServer` to build a WhatsApp bot application server with MindMeld.
+
+First, you need to install the specific dependencies for bot integration.
+
+.. code:: console
+
+   pip install mindmeld[bot]
+
+After that you can instantiate an WhatsAppBotServer instance. A sample implementation is provided in the HR blueprint.
+
+.. code:: python
+
+   import logging
+
+   from flask import Flask, request
+   from twilio.twiml.messaging_response import MessagingResponse
+
+   from mindmeld.components import NaturalLanguageProcessor
+   from mindmeld.components.dialogue import Conversation
+   from mindmeld import configure_logs
+
+
+   class WhatsappBotServer:
+       """
+       A sample server class for Whatsapp integration with any MindMeld application
+       """
+
+       def __init__(self, name, app_path, nlp=None):
+           """
+           Args:
+               name (str): The name of the server.
+               app_path (str): The path of the MindMeld application.
+               nlp (NaturalLanguageProcessor): MindMeld NLP component, will try to load from app path
+                 if None.
+           """
+           self.app = Flask(name)
+           if not nlp:
+               self.nlp = NaturalLanguageProcessor(app_path)
+               self.nlp.load()
+           else:
+               self.nlp = nlp
+           self.conv = Conversation(nlp=self.nlp, app_path=app_path)
+           self.logger = logging.getLogger(__name__)
+
+           @self.app.route("/", methods=["POST"])
+           def handle_message():  # pylint: disable=unused-variable
+               incoming_msg = request.values.get('Body', '').lower()
+               resp = MessagingResponse()
+               msg = resp.message()
+               response_text = self.conv.say(incoming_msg)[0]
+               msg.body(response_text)
+               return str(resp)
+
+       def run(self, host="localhost", port=7150):
+           self.app.run(host=host, port=port)
+
+
+   if __name__ == '__main__':
+       app = Flask(__name__)
+       configure_logs()
+       server = WhatsappBotServer(name='whatsapp', app_path='.')
+       port_number = 5000
+       print('Running server on port {}...'.format(port_number))
+       server.run(host='localhost', port=port_number)
+
+You can download the human resource blueprint for an example implementation under `whatsapp_bot_server.py`.
+
+.. code:: console
+
+   mindmeld blueprint hr_assistant
```

Comment:
Could you move these lines to where line 38 is? And merge this with that sentence "After that you can instantiate an WhatsAppBotServer instance. A sample implementation is provided in the HR blueprint."